portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,24 +34,46 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    static final String COMMENT_TABLE_NAME = "Comment";
+    static final String COMMENT_COLUMN_NAME = "comment";
+    static final String TIMESTAMP_COLUMN_NAME = "submit_time";
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Convert comments to JSON
-    List<String> comments = new ArrayList<>();
-    comments.add("This is amazing!");
-    comments.add("10/10!");
-    comments.add("LGTM");
-    String json = convertToJsonUsingGson(comments);
+      Query query = new Query(COMMENT_TABLE_NAME).addSort(TIMESTAMP_COLUMN_NAME, SortDirection.DESCENDING);
+      PreparedQuery results = datastore.prepare(query);
+      List<String> commentsList = new ArrayList<String>();
+      List<Entity> limitedResults = results.asList(FetchOptions.Builder.withLimit(2));
 
-    // Send JSON as response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+      for (Entity entity : limitedResults) {
+          String commentInQuery = (String) entity.getProperty(COMMENT_COLUMN_NAME);
+          commentsList.add(commentInQuery);
+      }
+    
+      response.setContentType("application/json");
+      String json = convertToJsonUsingGson(commentsList);
+      response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Gets comment input from <form> by id selector
+      String comment = request.getParameter("comment-input");
+      long submit_time = System.currentTimeMillis();
```

Comment:
No need to change anything now, but something to think about:
Using `System.currentTimeMillis()` will make this class difficult to test because it won't be deterministic. Let's chat about this offline.